operators/src/adapters/feature_collection_merger.rs

```diff
@@ -0,0 +1,250 @@
+///     Do we need an empty collection with column info as output instead?
+///     Do we put the columns to the stream's `VectorQueryContext` instead?
+#[pin_project(project = FeatureCollectionChunkMergerProjection)]
+pub struct FeatureCollectionChunkMerger<St, G>
+where
+    St: Stream<Item = Result<FeatureCollection<G>>> + FusedStream,
+    G: Geometry + ArrowTyped,
+{
+    #[pin]
+    stream: St,
+    accum: Option<FeatureCollection<G>>,
+    chunk_size_bytes: usize,
+}
+
+impl<St, G> FeatureCollectionChunkMerger<St, G>
+where
+    St: Stream<Item = Result<FeatureCollection<G>>> + FusedStream,
+    G: Geometry + ArrowTyped,
+{
+    pub fn new(stream: St, chunk_size_bytes: usize) -> Self {
+        Self {
+            stream,
+            accum: None,
+            chunk_size_bytes,
+        }
+    }
+
+    fn merge_and_proceed(
+        accum: &mut Option<FeatureCollection<G>>,
+        chunk_size_bytes: usize,
+        new_collection: St::Item,
+    ) -> Option<Poll<Option<St::Item>>> {
+        if new_collection.is_err() {
+            // TODO: maybe first output existing chunk and then the error?
+            return Some(Poll::Ready(Some(new_collection)));
+        }
+
+        let new_collection = new_collection.expect("checked");
+
+        let merged_collection = if let Some(old_collection) = accum.take() {
+            // TODO: execute on separate thread?
+            old_collection.append(&new_collection)
+        } else {
+            Ok(new_collection)
+        };
+
+        match merged_collection {
+            Ok(collection)
+                if !collection.is_empty() && collection.byte_size() >= chunk_size_bytes =>
+            {
+                Some(Poll::Ready(Some(Ok(collection))))
+            }
+            Ok(collection) => {
+                *accum = Some(collection);
+                None
+            }
+            Err(error) => Some(Poll::Ready(Some(Err(error.into())))),
+        }
+    }
+
+    fn output_remaining_chunk(accum: &mut Option<FeatureCollection<G>>) -> Poll<Option<St::Item>> {
+        match accum.take() {
+            Some(last_chunk) if !last_chunk.is_empty() => Poll::Ready(Some(Ok(last_chunk))),
+            _ => Poll::Ready(None),
+        }
+    }
+}
+
+impl<St, G> Stream for FeatureCollectionChunkMerger<St, G>
+where
+    St: Stream<Item = Result<FeatureCollection<G>>> + FusedStream,
+    G: Geometry + ArrowTyped,
+{
+    type Item = St::Item;
+
+    fn poll_next(mut self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Option<St::Item>> {
+        let FeatureCollectionChunkMergerProjection {
+            mut stream,
+            accum,
+            chunk_size_bytes,
+        } = self.as_mut().project();
+
+        let mut output: Option<Poll<Option<St::Item>>> = None;
+
+        while output.is_none() {
+            if stream.is_terminated() {
+                return Self::output_remaining_chunk(accum);
+            }
+
+            let next = ready!(stream.as_mut().poll_next(cx));
```

Comment:
does it really work when the result of `ready!`ist not `Ready(Ok(_))` ? The tests don't cover that case if I see it correctly

Reply:
ebe2168

Reply:
sorry for being imprecise here, I meant: what if the result is not Ready but pending?

Reply:
`ready!` returns early. As soon as the source calls the waker, this function will be called again and it returns `Ready` then.
How can I create a test that uses a stream that returns Pending for once without blocking…?
Is it even necessary? The logic is from `ready!` downwards.

Reply:
We need some stream that really produces results asynchronously to check the correct behavior. Something like my DelayOperator that somehow got deleted...

```
        self.points[0]
            .query(query, ctx)
            .then(async move |x| {
                let _ = tokio::spawn(async move {
                    thread::sleep(time::Duration::from_secs(seconds));
                })
                .await;
                x
            })
            .boxed()
    }
```

Reply:
I doubt that we should use delays in tests.
Could you create something that has your desired behavior without blocking with time?

Reply:
why should'nt we use delays in tests? Sleep does not use up cpu cycles, and as long as we don't sleep for multiple minutes it shouldn't be a problem for runtime. If we don't use sleep we need to constuct some interaction between two object and trigger the wake() externally which would be much more complicated.

Reply:
You will block up test threads (especially if `t=1`) with sleep.

93b6272

Reply:
I don't think so, because tokio::spawn uses a different thread pool than the testing framework

Reply:
Yes, but it would not spawn another test while the tests with the delays are blocking.